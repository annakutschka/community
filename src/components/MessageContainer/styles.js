import { StyleSheet } from 'react-native';
import Colors from '../../constants/Colors';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignItems: 'flex-start',
    margin: 10,
    padding: 10,
    width: '95%',
    height: 'auto',
    shadowColor: Colors.shadowColor,
    elevation: 3,
    backgroundColor: Colors.white
  },
  message: {
    flexDirection: 'column',
    margin: 10
  },
  text16: {
    fontSize: 16,
    paddingTop: 10,
    color: Colors.text
  },
  headline20: {
    alignItems: 'center',
    fontSize: 20,
    fontWeight: 'bold',
    color: Colors.text
  },
  textTime: {
    fontSize: 16,
    alignSelf: 'flex-end',
    color: Colors.bordergray
  },
  textBold16: {
    fontSize: 16,
    fontWeight: 'bold',
    color: Colors.text
  },
  containerUser: {
    width: '95%',
    flexDirection: 'row',
    margin: 5,
    justifyContent: 'space-between'
  },
  user: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start'
  },
  horizontalLine: {
    width: '95%',
    borderBottomColor: Colors.bordergray,
    borderBottomWidth: 1,
    margin: 5
  },
  profileImage: {
    flex: 1,
    borderRadius: 50,
    alignSelf: 'stretch',
    width: undefined,
    height: undefined
  },
  userName: {
    flex: 1,
    fontSize: 16,
    fontWeight: 'bold',
    color: Colors.text
  }
});

export default styles;
