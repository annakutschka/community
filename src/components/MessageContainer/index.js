import React from 'react';
import { Text, View, Image } from 'react-native';
import styles from './styles';

export default class MessageContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.message}>
          <Text style={styles.headline20}>{this.props.messageHeader} </Text>
          <Text style={styles.text16}>{this.props.messageText} </Text>
        </View>

        <View style={styles.horizontalLine} />

        <View style={styles.containerUser}>
          <View style={styles.user}>
            <Image
              source={this.props.messageUserImage}
              styles={styles.profilImage}
            />
            <Text style={styles.textBold16}>{this.props.messageUser} </Text>
          </View>
          <Text style={styles.textTime}>{this.props.messageTime} </Text>
        </View>
      </View>
    );
  }
}
