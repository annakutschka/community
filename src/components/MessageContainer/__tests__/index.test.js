import React from 'react';
import renderer from 'react-test-renderer';
import MessageContainer from '../index';

const props = {
  messageHeader: 'Title',
  messageText: 'Description',
  messageUser: 'User 1',
  messageTime: '12.03.14'
};

describe('<MessageContainer />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MessageContainer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
