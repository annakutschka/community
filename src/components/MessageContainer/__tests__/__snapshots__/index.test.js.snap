// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageContainer /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#fff",
      "elevation": 3,
      "flexDirection": "column",
      "height": "auto",
      "marginEnd": 15,
      "marginStart": 15,
      "marginTop": 10,
      "shadowColor": "#cccccc",
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "#3B3D3B",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Title
       
    </Text>
    <Text
      style={
        Object {
          "color": "#3B3D3B",
          "fontSize": 16,
          "paddingTop": 10,
        }
      }
    >
      Description
       
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#dedec2",
        "borderBottomWidth": 1,
        "margin": 5,
        "width": "95%",
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 10,
        "width": "95%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Image />
      <Text
        style={
          Object {
            "color": "#3B3D3B",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        User 1
         
      </Text>
    </View>
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "#dedec2",
          "fontSize": 16,
        }
      }
    >
      12.03.14
       
    </Text>
  </View>
</View>
`;
